packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -3517,4 +3517,97 @@ describe('TransactionController', () => {
       );
     });
   });
+
+  describe('updateTransactionHash', () => {
+    const transactionId = '1';
+    const hashMock = '1234';
+    const baseTransaction = {
+      id: transactionId,
+      chainId: toHex(5),
+      status: TransactionStatus.unapproved as const,
+      time: 123456789,
+      txParams: {
+        from: ACCOUNT_MOCK,
+        to: ACCOUNT_2_MOCK,
+      },
+    };
+    const transactionMeta: TransactionMeta = {
+      ...baseTransaction,
+      hash: hashMock,
+      history: [{ ...baseTransaction }],
+    };
+    it('updates transaction hash', async () => {
+      const newHash = 'newHash';
+      const controller = newController();
+      controller.state.transactions.push(transactionMeta);
+
+      controller.updateTransactionHash(transactionId, newHash);
+
+      const updatedTransaction = controller.state.transactions.find(
```

Comment:
Do we need to `find` in scenarios like this where there should only be a single transaction in the state?

Can we just use the first index?